### PR TITLE
Fix attachments in demo app

### PR DESCRIPTION
**Pull request by guperrot**
[AB#67101](https://dev.azure.com/msmobilecenter/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/67101)

### CODE REVIEW

File: demoapp/www/js/crashes.js [hunk 1]
```diff
@@ -119,15 +119,15 @@ $(document).bind('pageinit', function () {
         LowMemory.generateLowMemory();
     });
 
-    var updateAttachment = function () {
-        $('#text_attachment_value').html("Current value: " + text);
-        $('#file_attachment_value').html("Current value: " + attachment);
+    var updateAttachmentUI = function () {
+        $('#text_attachment_value').html("Current value: " + textAttachment);
```
Comment (michaelsmitha) [inline demoapp/www/js/crashes.js]:
```suggestion
        $('#text_attachment_value').text("Current value: " + textAttachment);
```

If the textAttachment is html, you'll be rendering the actual html in the dom.

File: demoapp/www/js/crashes.js [hunk 2]
```diff
@@ -151,15 +151,14 @@ $(document).bind('pageinit', function () {
...
-        }, function cameraError(error) {
-            alert("Unable to obtain picture: " + error);
+            binaryAttachment = imageUri;
+            attachmentsProvider.putString(attachmentsProvider.BINARY_KEY, binaryAttachment);
+            updateAttachmentUI();
+        }, function cameraError() {
+            binaryAttachment = null;
+            attachmentsProvider.putString(attachmentsProvider.BINARY_KEY, binaryAttachment);
+            updateAttachmentUI();
+
```
Comment (michaelsmitha) [inline demoapp/www/js/crashes.js]:
```suggestion
```

**Comment by guperrot** [comment]
Another issue is that the file on ios is temporary and works for only the next run for some reason.

File: demoapp/www/js/crashes.js [hunk 3]
```diff
@@ -151,15 +148,13 @@ $(document).bind('pageinit', function () {
...
-            attachment = imageUri.substr(0, index);
-            attachmentsProvider.putString(attachmentsProvider.BINARY_KEY, attachment);
-            updateAttachment();
-        }, function cameraError(error) {
-            alert("Unable to obtain picture: " + error);
+            binaryAttachment = imageUri;
+            attachmentsProvider.putString(attachmentsProvider.BINARY_KEY, binaryAttachment);
+            updateAttachmentUI();
+        }, function cameraError() {
+            binaryAttachment = null;
```
Comment (annakocheshkova) [inline demoapp/www/js/crashes.js]: I see that you're unsetting the value here but I would return the `alert` to be able to see what went wrong.

File: demoapp/www/js/crashes.js [hunk 4]
```diff
@@ -119,15 +112,19 @@ $(document).bind('pageinit', function () {
         LowMemory.generateLowMemory();
     });
 
-    var updateAttachment = function () {
-        $('#text_attachment_value').html("Current value: " + text);
-        $('#file_attachment_value').html("Current value: " + attachment);
+    var updateAttachmentUI = function () {
+        $('#text_attachment_value').text("Current value: " + textAttachment);
+        var imageDesc = null;
+        if (typeof (binaryAttachment) === "string") {
```
Comment (annakocheshkova) [inline demoapp/www/js/crashes.js]: Can it be not `string`? Is it null if we have no image stored? Should we then print `No image attached`?

Comment (annakocheshkova) [reply demoapp/www/js/crashes.js]: Btw if `binaryAttachemnt` contains teh uri why not just print it or some part of it?

**Comment by annakocheshkova** [comment]
@guperrot what do you mean by "file on ios is temporary"? I checked out the branch, attached the image, restarted the app - it persists.

File: demoapp/www/js/crashes.js (hunk 3 above)
Comment (guperrot) [reply demoapp/www/js/crashes.js]: Clicking cancel would display an error. On other platforms we cancel the dialog to unset attachment too.
You get an error at restart still.

File: demoapp/www/js/crashes.js (hunk 4 above)
Comment (guperrot) [reply demoapp/www/js/crashes.js]: This is base64 data and this check is to avoid checking for both null and undefined while being strict about the type.

**Comment by guperrot** [comment]
> @guperrot what do you mean by "file on ios is temporary"? I checked out the branch, attached the image, restarted the app - it persists.

No it works only 1 time for me, 100% repro.

**Pull request merged**